class-twicpics-urls-manager.php

```diff
@@ -0,0 +1,467 @@
+<?php
+/**
+ * TwicPics URLs Manager
+ *
+ * @package TwicPics
+ */
+
+defined( 'ABSPATH' ) || die( 'ERROR !' );
+
+/**
+ * URL constants as defined in the PHP Manual under "Constants usable with
+ * http_build_url()".
+ *
+ * @see http://us2.php.net/manual/en/http.constants.php#http.constants.url
+ */
+if ( ! defined( 'HTTP_URL_REPLACE' ) ) {
+	define( 'HTTP_URL_REPLACE', 1 );
+}
+if ( ! defined( 'HTTP_URL_JOIN_PATH' ) ) {
+	define( 'HTTP_URL_JOIN_PATH', 2 );
+}
+if ( ! defined( 'HTTP_URL_JOIN_QUERY' ) ) {
+	define( 'HTTP_URL_JOIN_QUERY', 4 );
+}
+if ( ! defined( 'HTTP_URL_STRIP_USER' ) ) {
+	define( 'HTTP_URL_STRIP_USER', 8 );
+}
+if ( ! defined( 'HTTP_URL_STRIP_PASS' ) ) {
+	define( 'HTTP_URL_STRIP_PASS', 16 );
+}
+if ( ! defined( 'HTTP_URL_STRIP_AUTH' ) ) {
+	define( 'HTTP_URL_STRIP_AUTH', 32 );
+}
+if ( ! defined( 'HTTP_URL_STRIP_PORT' ) ) {
+	define( 'HTTP_URL_STRIP_PORT', 64 );
+}
+if ( ! defined( 'HTTP_URL_STRIP_PATH' ) ) {
+	define( 'HTTP_URL_STRIP_PATH', 128 );
+}
+if ( ! defined( 'HTTP_URL_STRIP_QUERY' ) ) {
+	define( 'HTTP_URL_STRIP_QUERY', 256 );
+}
+if ( ! defined( 'HTTP_URL_STRIP_FRAGMENT' ) ) {
+	define( 'HTTP_URL_STRIP_FRAGMENT', 512 );
+}
+if ( ! defined( 'HTTP_URL_STRIP_ALL' ) ) {
+	define( 'HTTP_URL_STRIP_ALL', 1024 );
+}
+
+class TwicPicsUrlsManager {
+
+	public function __construct() {
+		$this->_website_url = ( isset( $_SERVER['HTTPS'] ) && 'on' === $_SERVER['HTTPS'] ? "https" : "http" ) . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]";
+	}
+
+	/**
+	 * Provides the ability to write logs in JavaScript.
+	 * 
+	 * @param      string $log The log to write
+	 */
+	private function write_logs( $log ) {
```

Comment:
Strange to put that in TwicPicsUrlsManager. Not a blocker for a merge but let's put the idea of moving it to another class entirely later on.

Reply:
Completely agree. I defined this function here for now, in case we need to use it in this class. But of course, we'll move it to a dedicated utility class later.